service/utils/tds.py

```diff
@@ -218,6 +271,18 @@ def attach_files(output: dict, job_id, status="complete"):
     if data_result is not None:
         data_result.to_csv(output_filename, index=False)
         files[output_filename] = "result.csv"
+        # Add a result summary file for the HMI to digest.
+        try:
+            result_summary_filename = os.path.join(job_dir, "./result_summary.csv")
+            summary_data = get_result_summary(data_result)
+            summary_df = pd.DataFrame.from_records(summary_data)
+            summary_df.to_csv(result_summary_filename)
+            files[result_summary_filename] = "result_summary.csv"
```

Comment:
I think you are grouping on the wrong dimensions, also why don't we just use the internal Pandas functions to do the statistics instead of rolling them ourselves? 

Maybe something like this?

```
>>>
>>> df = pd.DataFrame({
...     "Sample": [1, 1, 2, 2],
...     "Step": [1, 2, 1, 2],
...     "beta": [45000, 70000, 40000, 55000],
...     "gamma": [6, 15, 1, 12]
... })
>>>
>>> df2 = df.groupby("Step", as_index=False).agg(["mean", "max"])
>>> df2 = df2.drop("Sample", axis=1)
<stdin>:1: PerformanceWarning: dropping on a non-lexsorted multi-index without a level parameter may impact performance.
>>> df2.columns=["_".join(i) for i in df2.columns]
>>>
>>> df2
   Step_  beta_mean  beta_max  gamma_mean  gamma_max
0      1    42500.0     45000         3.5          6
1      2    62500.0     70000        13.5         15
```